### ISSUE TITLE
[CT-2074] [Bug] Error message for exposures gives incorrect node type for TargetNotFound

### ISSUE THREAD

**Issue by callum-mcdata** [dbt-labs/dbt-core#6928]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Assume I have the following exposure:

```yml
version: 2

exposures:

  - name: weekly_metrics
    label: Some Label
    type: dashboard
    maturity: high
    url: https://bi.tool/dashboards/1
    description: >
      Did someone say "exponential growth"?

    depends_on:
      - ref('fact_orders')
      - ref('dim_customers')
      - metric('testing')

    owner:
      name: Callum McData
      email: data@jaffleshop.com
```

But I the node `metric('testing')` does not actually exist in my project. If this is the case, I get the following error message when trying to compile:

> Compilation Error
  Exposure 'exposure.dbt_metrics_integration_tests.weekly_metrics' (models/materialized_models/exposures.yml) depends on a source named 'testing' which was not found

This is incorrect because we can clearly see that this is a `metric` and not a source. When trying out the same functionality with a model node, I get the correct type displayed in the error message. IE:
> depends on a node 'testing' which was not found

### Expected Behavior

I would expect that the CompilationError returned by dbt core would identify the correct target node type. However, for the life of me I cannot figure out where this is coming from. I am fairly certain the error message is arising from `TargetNotFoundError` but I was unable to find the path that routed into there with this issue.

### Steps To Reproduce

1. Create an exposure that is referencing a metric that doesn't exist

### Relevant log output

```shell
Shown in earlier messages
```

### Environment

```markdown
- OS: MacOS
- Python: 3.10
- dbt: 1.4.1
```

### Which database adapter are you using with dbt?

postgres

**Comment by jtcohen6** [comment]
@callum-mcdata Thanks for opening! I think this is a one-liner:
https://github.com/dbt-labs/dbt-core/blob/311a57a21e61955904202bdb469c2077202e9073/core/dbt/parser/manifest.py#L1260-L1271

`target_kind` should be `"metric"`, not `"source"`!

**Comment by callum-mcdata** [comment]
@jtcohen6 yep - that's the smoking bullet right there! Tested it out on my local and it resolved the issue. Fixed via PR #6934
